Networking/Networking/Model/Product/ProductPreviewGenerator.swift

```diff
@@ -0,0 +1,144 @@
+import Foundation
+
+extension URL {
+    func appendingHideMasterbarParameters() -> URL? {
+        guard let components = URLComponents(url: self, resolvingAgainstBaseURL: false) else {
+            return nil
+        }
+        // FIXME: This code is commented out because of a menu navigation issue that can occur while
+        // viewing a site within the webview. See https://github.com/wordpress-mobile/WordPress-iOS/issues/9796
+        // for more details.
+        //
+        // var queryItems = components.queryItems ?? []
+        // queryItems.append(URLQueryItem(name: "preview", value: "true"))
+        // queryItems.append(URLQueryItem(name: "iframe", value: "true"))
+        // components.queryItems = queryItems
+        /////
+        return components.url
+    }
+}
+
+public final class ProductPreviewGenerator: NSObject {
+    let product: Product
+    private let frameNonce: String?
+    fileprivate let authenticator: WebViewAuthenticator?
```

Comment:
`fileprivate` is gone for good!

Reply:
Second note: Since `WebViewAuthenticator` doesn't have a vailable initializer, can we remove the `?` ?

That would allow you to get rid of several `guard let` all over :fire: